src/main/scala/viper/gobra/translator/encodings/StringEncoding.scala

```diff
@@ -68,13 +75,61 @@ class StringEncoding extends LeafTypeEncoding {
           lowExp  <- goE(low)
           highExp <- goE(high)
         } yield withSrc(vpr.FuncApp(strSlice, Seq(baseExp, lowExp, highExp)), slice)
+      case conv@in.Conversion(in.StringT(_), expr :: ctx.Slice(in.IntT(_, TypeBounds.Byte))) =>
+        val (pos, info, errT) = conv.vprMeta
+        for { e <- goE(expr) } yield byteSliceToStr(e)(ctx)(pos, info, errT)
+    }
+  }
+
+
+  /**
+    * Encodes the (effectul) conversion from a string to a []byte
+    * [ target = []byte(str) ] ->
+    *   [
+    *     var s []byte
+    *     inhale forall i Int :: 0 <= i && i < len(s) ==> acc(&s[i])
```

Comment:
is it correct that you (currently) only take care of the permissions without restricting the resulting values in any way?

Reply:
Yes, that is the point. We can enrich the specification once we introduce more string operations (e.g. indexing strings)